core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -1474,8 +1470,39 @@ class TimeSeriesShard(val ref: DatasetRef,
     }
   }
 
+  /**
+    * Note this approach below assumes the following for quick size estimation. The sizing is more
+    * a swag than reality:
+    * (a) every matched time series ingests at all query times. Looking up start/end times and more
+    *     precise size estimation is costly
+    * (b) it also assigns bytes per sample based on schema which is much of a swag. In reality, it would depend on
+    *     number of histogram buckets, samples per chunk etc.
+    * (c) Assumes reporting interval is 20s.
+    */
+  private def capDataScannedPerShardCheck(lookup: PartLookupResult) = {
+    lookup.firstSchemaId.foreach { schId =>
+      lookup.chunkMethod match {
+        case TimeRangeChunkScan(st, end) =>
+          val numMatches = lookup.partsInMemory.length + lookup.partIdsNotInMemory.length
```

Comment:
Could we move the code into a common place as it's repeated in both Shard classes?  Also, could we put the assumed resolution somewhere as a constant so we can easily change it if needed?